### PR TITLE
Feature/e structure

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/document/hooks/data/create/section-columns-limit.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import HookDependency from '../base/dependency';
+import DataDependency from 'elementor-api/modules/hooks/data-base/dependency';
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/data/create/section-columns-limit.js]:
```suggestion
import DataDependency from 'elementor-api/modules/hooks/data/dependency-base';
```

File: assets/dev/js/editor/document/hooks/data/index.js [hunk 2]
```diff
@@ -4,5 +4,4 @@ export * from './create/';
 export * from './delete/';
 export * from './move/';
 export * from './paste/';
-export * from './paste-style/';
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/data/index.js]: Where is it?

File: assets/dev/js/editor/document/hooks/ui/create/column-is-populated.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import EventAfter from '../base/after';
+import UIAfter from 'elementor-api/modules/hooks/ui-base/after';
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/create/column-is-populated.js]:
```suggestion
import UIAfter from 'elementor-api/modules/hooks/ui/after-base';
```

File: assets/dev/js/editor/elements/views/section.js [hunk 4]
```diff
@@ -13,8 +13,7 @@ const SectionView = BaseElementView.extend( {
 
 	addSectionView: null,
 
-	_checkIsFull: function() {
-		// TODO: should be part of $e.events.
+	toggleSectionIsFull: function() {
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/section.js]: BC

File: core/common/assets/js/api/api/commands.js [hunk 5]
```diff
@@ -12,10 +16,30 @@ export default class Commands extends elementorModules.Module {
 		this.components = {};
 	}
 
+	/**
+	 * Function getAll().
+	 *
+	 * Receive all loaded commands.
+	 *
+	 * @notice List of command available [here](method---ecommands-get-all.md)
```
Comment (matipojo) [inline core/common/assets/js/api/api/commands.js]: Currently remove any reference to md docs.

File: core/common/assets/js/api/api/hooks.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
+import DataHooks from './hooks/data.js';
```
Comment (matipojo) [inline core/common/assets/js/api/api/hooks.js]: Rename

File: core/common/assets/js/api/api/hooks/base.js [hunk 7]
```diff
@@ -117,24 +117,22 @@ export default class Callbacks extends elementorModules.Module {
 		const { containers = [ args.container ] } = args,
 			elType = containers[ 0 ] ? containers[ 0 ].type : false;
```
Comment (matipojo) [inline core/common/assets/js/api/api/hooks/base.js]: elType => containerType

File: core/common/assets/js/api/api/hooks/base.js [hunk 8]
```diff
@@ -156,11 +154,11 @@ export default class Callbacks extends elementorModules.Module {
 	 *
 	 * Validate given instance.
 	 *
-	 * @param {CallbackBase} instance
+	 * @param {HookBase} instance
 	 */
 	checkInstance( instance ) {
 		if ( instance.getType() !== this.getType() ) {
-			throw new Error( `invalid instance, please use: 'elementor-document/callback/base/base'. ` );
+			throw new Error( `invalid instance, please use: 'elementor-api/modules/callable-base'. ` );
```
Comment (matipojo) [inline core/common/assets/js/api/api/hooks/base.js]: `callable`?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/hooks/data/index.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/hooks/data/index.js]: Not needed.

File: core/common/assets/js/api/api/commands.js [hunk 9]
```diff
@@ -1,7 +1,11 @@
-/**
- * TODO: Full JSDOC.
- */
 export default class Commands extends elementorModules.Module {
+	/**
```
Comment (matipojo) [inline core/common/assets/js/api/api/commands.js]: Rename api/api

File: core/common/assets/js/modules.js [hunk 10]
```diff
@@ -1,13 +1,13 @@
 import elementorModules from '../../../../assets/dev/js/modules/modules';
 import ModalLayout from './views/modal/layout';
-import Component from './components/component';
-import ComponentModal from './components/component-modal';
-import HookBreak from './components/hook-break';
+import Component from './api/modules/component-base';
+import ComponentModal from './api/modules/component-modal';
```
Comment (matipojo) [inline core/common/assets/js/modules.js]:
```suggestion
import ComponentModal from './api/modules/component-modal-base';
```

File: core/common/assets/js/modules.js [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
+import HookBreak from './api/modules/hook-break';
 
 elementorModules.common = {
-	Component: Component,
-	ComponentModal: ComponentModal,
-	HookBreak: HookBreak,
+	Component: Component, // TODO: remove use `$e.modules.ComponentBase`.
```
Comment (matipojo) [inline core/common/assets/js/modules.js]: Add deprecated notice. for example:
https://github.com/elementor/elementor/blob/5ee95138703488fd7c82a4a0939c801d1718cbc1/assets/dev/js/editor/editor-base.js#L1079
But use `get()` instehad of `defineProperty`.

File: tests/qunit/core/editor/document/ui/commands/copy.spec.js [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import ElementsHelper from '../../elements/helper';
-import CommonHelper from '../../../../common/components/helper';
+import CommonHelper from '../../../../common/helper';
```
Comment (matipojo) [inline tests/qunit/core/editor/document/ui/commands/copy.spec.js]: Use an alias

File: tests/qunit/core/editor/document/ui/commands/delete.spec.js [hunk 13]
```diff
@@ -1,5 +1,5 @@
 import ElementsHelper from '../../elements/helper';
-import CommonHelper from '../../../../common/components/helper';
+import CommonHelper from '../../../../common/helper';
```
Comment (matipojo) [inline tests/qunit/core/editor/document/ui/commands/delete.spec.js]: Use an alias, here and every `../../`

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

**Pull request merged**